### PR TITLE
[google_maps_flutter]: LatLng longitude loses precision in constructor #90574

### PR DESCRIPTION
**Pull request by Murray-Meller**
This PR helps avoid unnecessary loss of longitude precision in the `LatLng` constructor. 

See issue for more details.

## Issues:
Fixes https://github.com/flutter/flutter/issues/90574

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Comment by Murray-Meller** [comment]
@googlebot I signed it!

**Comment by MurrayMeller** [comment]
@googlebot I consent.

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/location.dart [hunk 1]
```diff
@@ -14,13 +14,16 @@ class LatLng {
   /// The latitude is clamped to the inclusive interval from -90.0 to +90.0.
   ///
   /// The longitude is normalized to the half-open interval from -180.0
-  /// (inclusive) to +180.0 (exclusive)
+  /// (inclusive) to +180.0 (exclusive). Avoids normalization if possible to
+  /// avoid unnecessary loss of precision
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/location.dart:18]: The fact that no operations are performed on it if it's already in range is purely an implementation detail, so doesn't belong in the documentation comment. You can move this down to be a normal comment on line 24 instead, to explain the conditional.

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/test/types/location_test.dart [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:google_maps_flutter_platform_interface/google_maps_flutter_platform_interface.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  group('LanLng constructor', () {
+    test('Maintains longitude precision  if within acceptable range', () async {
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/test/types/location_test.dart:12]: Nit: fix double space before "if"

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/location.dart (hunk 1 above)
Comment (Murray-Meller) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/location.dart:18]: Good point. I like that reasoning. I'll update it

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/test/types/location_test.dart (hunk 2 above)
Comment (Murray-Meller) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/test/types/location_test.dart]: Thanks. I'll update

**Comment by Murray-Meller** [comment]
Thanks for the review, @stuartmorgan. I have addressed both your comments. 

I was wondering if you'd be able to help me understand more about my pipeline failures, where I could look and what I could do to get them fixed?

**Comment by ditman** [comment]
> Incase it changes, these checks were failing:
>
>  ci.yaml validation
>  submit-queue

I can explain:

* `submit-queue` tracks the status of tests of the commits in the master branch of flutter/plugins. If the tests don't pass in master, other PRs are blocked, until master recovers.
* `ci.yaml validation` validates that the config for one of our continuous integration systems (LUCI) is OK. The case why it fails is because of a transient error, or because the branch was forked off at a point in master where the test was broken (and a fix was missing). _(I checked this branch, and I didn't see any changes related to that validation, so it's probably a transient failure.)_

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
Adding @ditman for secondary review per Flutter policy.

**Review by ditman — APPROVED** [review]
LGTM!

**Comment by MurrayMeller** [comment]
Thank again, @ditman. Much appreciated.

**Comment by ditman** [comment]
@MurrayMeller thanks again for your contribution!! Keep them coming!

**Pull request merged**